database/sqlite/sqlite_context.c

```diff
@@ -0,0 +1,601 @@
+    snprintfz(buf, 1024, "PRAGMA journal_mode=%s;", config_get(CONFIG_SECTION_SQLITE, "journal mode", "WAL"));
+    if(init_database_batch(db_context_meta, DB_CHECK_NONE, 0, list)) return 1;
+
+    // https://www.sqlite.org/pragma.html#pragma_temp_store
+    // PRAGMA temp_store = 0 | DEFAULT | 1 | FILE | 2 | MEMORY;
+    snprintfz(buf, 1024, "PRAGMA temp_store=%s;", config_get(CONFIG_SECTION_SQLITE, "temp store", "MEMORY"));
+    if(init_database_batch(db_context_meta, DB_CHECK_NONE, 0, list)) return 1;
+    
+    // https://www.sqlite.org/pragma.html#pragma_journal_size_limit
+    // PRAGMA schema.journal_size_limit = N ;
+    snprintfz(buf, 1024, "PRAGMA journal_size_limit=%lld;", config_get_number(CONFIG_SECTION_SQLITE, "journal size limit", 16777216));
+    if(init_database_batch(db_context_meta, DB_CHECK_NONE, 0, list)) return 1;
+
+    // https://www.sqlite.org/pragma.html#pragma_cache_size
+    // PRAGMA schema.cache_size = pages;
+    // PRAGMA schema.cache_size = -kibibytes;
+    snprintfz(buf, 1024, "PRAGMA cache_size=%lld;", config_get_number(CONFIG_SECTION_SQLITE, "cache size", -2000));
+    if(init_database_batch(db_context_meta, DB_CHECK_NONE, 0, list)) return 1;
+
+    snprintfz(buf, 1024, "PRAGMA user_version=%d;", target_version);
+    if(init_database_batch(db_context_meta, DB_CHECK_NONE, 0, list)) return 1;
+
+    snprintfz(buf, 1024, "ATTACH DATABASE \"%s/netdata-meta.db\" as meta;", netdata_configured_cache_dir);
+    if(init_database_batch(db_context_meta, DB_CHECK_NONE, 0, list)) return 1;
+
+    if (init_database_batch(db_context_meta, DB_CHECK_NONE, 0, &database_context_config[0]))
+        return 1;
+
+    if (init_database_batch(db_context_meta, DB_CHECK_NONE, 0, &database_context_cleanup[0]))
+        return 1;
+
+    return 0;
+}
+
+/*
+ * Close the sqlite database
+ */
+
+void sql_close_context_database(void)
+{
+    int rc;
+    if (unlikely(!db_context_meta))
+        return;
+
+    info("Closing context SQLite database");
+
+    rc = sqlite3_close_v2(db_context_meta);
+    if (unlikely(rc != SQLITE_OK))
+        error_report("Error %d while closing the context SQLite database, %s", rc, sqlite3_errstr(rc));
+    return;
+}
+
+//
+// Fetching data
+//
+#define CTX_GET_CHART_LIST  "SELECT c.chart_id, c.type||'.'||c.id, c.name, c.context, c.title, c.unit, c.priority, " \
+        "c.update_every, c.chart_type, c.family FROM meta.chart c WHERE c.host_id = @host_id; "
+
+void ctx_get_chart_list(uuid_t *host_uuid, void (*dict_cb)(SQL_CHART_DATA *, void *), void *data)
+{
+    int rc;
+    sqlite3_stmt *res = NULL;
+
+    if (unlikely(!host_uuid)) {
+       internal_error(true, "Requesting context chart list without host_id");
+       return;
+    }
+
+    rc = sqlite3_prepare_v2(db_context_meta, CTX_GET_CHART_LIST, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to fetch chart list");
+        return;
+    }
+
+    if (likely(host_uuid))
```

Comment:
It is always true because of lines 131-134

Reply:
Removed